### ISSUE TITLE
Custom tag for b-dropdown

### ISSUE THREAD

**Issue by akaNightmare** [bootstrap-vue/bootstrap-vue#533]
Right now it's alway a button https://github.com/bootstrap-vue/bootstrap-vue/blob/master/lib/components/dropdown.vue, but it would be great to use something like:
```html
<div class="user__info" data-toggle="dropdown">
    <div>
        <div class="user__name">Test test</div>
        <div class="user__email">test@test.com</div>
   </div>
</div>
```

Thanks.

**Comment by tmorehouse** [comment]
For clarity, are you referring to the button that toggles the drop-down menu visible state?

**Comment by tmorehouse** [comment]
Hmm I wonder if hte you could use the button `link` variant, which removes the look up a button, but keeps full functionality.

**Comment by akaNightmare** [comment]
@tmorehouse 
> For clarity, are you referring to the button that toggles the drop-down menu visible state?
yes
> Hmm I wonder if the you could use the button link variant, which removes the look of a button, but keeps full functionality, and then you could place your two inner divs in the button slot.
I am not sure, what you mean

**Comment by tmorehouse** [comment]
Try this out:

```html
  <b-dropdown variant="link">
    <template slot="text">
      <div>
        <div class="user__name">Test test</div>
        <div class="user__email">test@test.com</div>
      </div>
    </template>
    <b-dropdown-header>This is a heading</b-dropdown-header>
    <b-dropdown-item>Action</b-dropdown-item>
    <b-dropdown-item>Another action</b-dropdown-item>
    <b-dropdown-divider></b-dropdown-divider>
    <b-dropdown-item>Something else here...</b-dropdown-item>
  </b-dropdown>
```

**Comment by tmorehouse** [comment]
Be default the "caret" symbol gets placed at the bottom because the `div` are block level element, rather than the side, so you may need to float your content to the left (`float-left`), or change the divs to spans

**Comment by akaNightmare** [comment]
@tmorehouse yep, I saw that example, but root element still `button`

**Comment by tmorehouse** [comment]
Yeah, currently `b-dropdown` requires a button.  I'm not sure about the extent of the changes that would be needed in the component to handle a non-button (including the caret stuff)

**Comment by tmorehouse** [comment]
, although the above example will have the appearance of a link, rather than a button (although it still has the padding of a button).

This might be something we have investigate for use when bootstrap v4 beta is released, as they are making changes to how dropdowns are being handled (they will be switching to popper.js to handle positioning of the dropdowns)... so major changes coming along.

**Comment by akaNightmare** [comment]
@tmorehouse thanks a lot for a fast feedbacks)
